src/odemis/driver/lakeshore.py

```diff
@@ -502,9 +501,18 @@ def _parseMessage(self, msg):
             vals = msg[5:].split(',')
             self._heating = int(vals[1])
         # Query temperature
-        elif re.match(b'SRDG\?', msg):
-            self._sendAnswer(b"%f" % (self._temperature,))
-            self._temperature += random.randint(-5, 5)
+        elif re.match(b'KRDG\?', msg):
+            # send temperature with some noise
+            self._sendAnswer(b"+%.3f" % (self._temperature + random.random() / 1000,))
+            if self._heating:
+                if self._temperature < self._setpoint:  # heating is enabled
+                    self._temperature += 0.05 * self._heating  # simulate heating
+                else:
+                    self._temperature -= 0.01
+            else:  # no heating so no temperature control
+                # maintain stable temperature
+                if self._temperature > self._stable_temperature:
+                    self._temperature -= 0.1  # cool off with no heating
```

Comment:
This is a little odd with the simulator as the temperature rate depends on how frequently you're reading it. That said, for now that should be enough... so let's keep it like this.